resample_spatial.json

```diff
@@ -12,7 +12,16 @@
             "description": "A raster data cube.",
             "schema": {
                 "type": "object",
-                "subtype": "raster-cube"
+                "subtype": "datacube",
+                "dimensions": [
+                    {
+                        "type": "spatial",
```

Comment:
Couldn't this also apply to vector cubes? Not for resampling ofc, but for reprojection!
Apologies if this would be obvious to someone with more EO background than I have!

Reply:
That is true and makes things complicated, unfortunately. Maybe this should be a separate process though as the behavior is really quite a bit different with one working just on the vector labels?!

Reply:
Hmm - is the behaviour really different enough to require entirely separate processes? The `resolution` parameter (which wouldn't apply to vector cubes) is already optional and the resampling methods would also apply to both, right? In the spirit of minimalism (or as a user) I think I favour for this to remain one process that does both.

I also just realised that I'd like to extend my comment to `resample_cube_spatial` - I think there it could be quite useful to allow all datacubes - e.g. to reproject a rastercube to the CRS of a vectorcube.

Reply:
I don't think the `method` and `align` parameters apply to vector. I could see a use case for having both supported in `resample_cube_spatial` as you may want to pass a raster and vector cube for alignment, but for resample_spatial the vector variant would be much simpler and also usually be a "different code path". What do others think?
A new process `resample_vector` could be defined as follows:
```json
{
    "id": "resample_vector",
    "summary": "Reprojects the vector dimension",
    "description": "Converts the geometries stored in the vector dimension to a different projection.\n\nTo spatially align two data cubes with each other, better use the process ``resample_cube_spatial()``.",
    "categories": [
        "cubes",
        "aggregate & resample",
        "vector"
    ],
    "parameters": [
        {
            "name": "data",
            "description": "A vector data cube.",
            "schema": {
                "type": "object",
                "subtype": "datacube",
                "dimensions": [
                    {
                        "type": "vector"
                    }
                ]
            }
        },
        {
            "name": "projection",
            "description": "Reprojects the geometries stored in the vector dimension to the given projection, specified as as [EPSG code](http://www.epsg-registry.org/), [WKT2 (ISO 19162) string](http://docs.opengeospatial.org/is/18-010r7/18-010r7.html), [PROJ definition (deprecated)](https://proj.org/usage/quickstart.html).",
            "schema": [
                {
                    "title": "EPSG Code",
                    "type": "integer",
                    "subtype": "epsg-code",
                    "minimum": 1000,
                    "examples": [
                        3857
                    ]
                },
                {
                    "title": "WKT2",
                    "type": "string",
                    "subtype": "wkt2-definition"
                }
            ]
        }
    ],
    "returns": {
        "description": "A vector data cube with geometries converted into the new projection. The reference system of the vector dimension changes, all other dimensions and properties remain unchanged.",
        "schema": {
            "type": "object",
            "subtype": "datacube",
            "dimensions": [
                {
                    "type": "vector"
                }
            ]
        }
    },
    "links": [
        {
            "href": "https://openeo.org/documentation/1.0/datacubes.html#resample",
            "rel": "about",
            "title": "Resampling explained in the openEO documentation"
        },
        {
            "rel": "about",
            "href": "https://proj.org/usage/projections.html",
            "title": "PROJ parameters for cartographic projections"
        },
        {
            "rel": "about",
            "href": "http://www.epsg-registry.org",
            "title": "Official EPSG code registry"
        },
        {
            "rel": "about",
            "href": "http://www.epsg.io",
            "title": "Unofficial EPSG code database"
        }
    ]
}

```